test/relayserver/RelayServer.test.ts

```diff
@@ -666,6 +699,199 @@ describe('RelayServer', function () {
           )
         ).to.be.fulfilled;
       });
+
+      describe('with boltz verifier', function () {
+        let swap: TestSwap;
+        let index = 1;
+
+        beforeEach(async function () {
+          swap = await deployContract<TestSwap>('TestSwap');
+          encodedData = swap.interface.encodeFunctionData('claim', [
+            constants.HashZero,
+            ethers.utils.parseEther('0.5'),
+            constants.AddressZero,
+            500,
+          ]);
+          await boltzVerifier.acceptContract(swap.address);
+          index++;
+        });
+
+        it('should relay deploy transaction with contract execution', async function () {
+          const [fundedAccount] = (await ethers.getSigners()) as [
+            SignerWithAddress
+          ];
+          await fundedAccount.sendTransaction({
+            to: swap.address,
+            value: ethers.utils.parseEther('1'),
+          });
+          const userDefinedRelayRequest = createUserDefinedRequest(
+            true,
+            {
+              from: owner.address,
+              to: swap.address,
+              data: encodedData,
+              index,
+            },
+            {
+              callForwarder: boltzFactory.address,
+              callVerifier: boltzVerifier.address,
+            }
+          );
+
+          const envelopingTxRequest = await createEnvelopingTxRequest(
+            userDefinedRelayRequest,
+            relayClient,
+            hubInfo
+          );
+
+          await expect(
+            relayServer.createRelayTransaction(
+              stringifyEnvelopingTx(envelopingTxRequest)
+            )
+          ).to.be.fulfilled;
+        });
+
+        it('should fail if verifier throws error', async function () {
+          const userDefinedRelayRequest = createUserDefinedRequest(
+            true,
+            {
+              from: owner.address,
+              to: swap.address,
+              data: encodedData,
+              tokenGas: 5000,
+              tokenAmount: ethers.utils.parseEther('5'),
+              index,
+            },
+            {
+              callForwarder: boltzFactory.address,
+              callVerifier: boltzVerifier.address,
+            }
+          );
+
+          const envelopingTxRequest = await createEnvelopingTxRequest(
+            userDefinedRelayRequest,
+            relayClient,
+            hubInfo
+          );
+
+          await expect(
+            relayServer.createRelayTransaction(
+              stringifyEnvelopingTx(envelopingTxRequest)
+            )
+          ).to.be.rejectedWith('Claiming value lower than fees');
+        });
+
+        // TODO - Should bubble up error but its failing
+        it.skip('should fail if destination contract throws error', async function () {
```

Comment:
In this case, I think that the good part is that still the promise rejects as expected, but we don't get the proper error message. I'd try to investigate a bit more on the reason, otherwise, I wouldn't skip the test, but I'd add a comment on the expected error clarifying it.

Reply:
correct it fails but with a `transaction reverted`, let me fix it then. 

We already know the issue and how to fix it, we discuss fixing in a different task.

Reply:
updated